src/integrationTest/groovy/wooga/gradle/unity/UnityTaskIntegrationTest.groovy

```diff
@@ -0,0 +1,255 @@
+        method                    | rawValue         | type                      | append | expectedValue
+        "argument"                | "--foo"          | "String"                  | true   | ["--test", "value", "--foo"]
+        "arguments"               | ["--foo", "bar"] | "List<String>"            | true   | ["--test", "value", "--foo", "bar"]
+        "arguments"               | ["--foo", "bar"] | "String[]"                | true   | ["--test", "value", "--foo", "bar"]
+        "setAdditionalArguments"  | ["--foo", "bar"] | "List<String>"            | false  | ["--foo", "bar"]
+        "setAdditionalArguments"  | ["--foo", "bar"] | "Provider<List<String>>"  | false  | ["--foo", "bar"]
+        "additionalArguments.set" | ["--foo", "bar"] | "List<String>"            | false  | ["--foo", "bar"]
+        "additionalArguments.set" | ["--foo", "bar"] | "Provider<List<String>>>" | false  | ["--foo", "bar"]
+
+        property = "additionalArguments"
+        value = wrapValueBasedOnType(rawValue, type)
+        message = (append) ? "which appends arguments" : "which replaces arguments"
+    }
+
+    @Unroll
+    def "can set unityPath with #method"() {
+        given: "a build file with custom test task"
+        appendToTestTask("$method(file('$value'))")
+
+        and:
+        def testUnity = createFile(value)
+        Files.copy(mockUnityFile.toPath(), testUnity.toPath(), StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.COPY_ATTRIBUTES)
+
+        when:
+        def result = runTasksSuccessfully(testTaskName)
+
+        then:
+        if (isWindows) {
+            value = value.replace('/', '\\')
+        }
+        result.standardOutput.contains(value)
+
+        where:
+        property    | useSetter | value
+        "unityPath" | true      | "custom/unity.bat"
+        "unityPath" | false     | "custom/unity.bat"
+
+        method = (useSetter) ? "set${property.capitalize()}" : "${property}.set"
+    }
+
+    @Unroll
+    def "set log category with #method to '#value'"() {
+        given:
+        buildFile << """
+            ${testTaskName}.$method(${value})
+            """.stripIndent()
+        addProviderQueryTask("custom", "${testTaskName}.unityLogFile", ".get().asFile.path")
+
+        when:
+        def result = runTasksSuccessfully(testTaskName, "custom")
+
+        then:
+        result.wasExecuted(testTaskName)
+        def resultPath = new File(projectDir, "/build/logs/${path}").getPath()
+        //result.standardOutput.contains("-logFile ${resultPath}")
+        result.standardOutput.contains("${testTaskName}.unityLogFile: ${resultPath}")
+
+        where:
+        rawValue     | useSetter | path
+        "helloworld" | true      | "helloworld/${testTaskName}.log"
+        "helloworld" | false     | "helloworld/${testTaskName}.log"
+        "foobar"     | true      | "foobar/${testTaskName}.log"
+        ""           | true      | "${testTaskName}.log"
+        ""           | false     | "${testTaskName}.log"
```

Comment:
These two examples error on macOS because the logfile path it prodocues is `/${testTaskName}.log` which is not writeable. I couldn't find the logic where the full path gets combined.